pom.xml

```diff
@@ -75,7 +75,7 @@
         <mrunit.version>1.1.0</mrunit.version>
         <mockito.version>2.23.0</mockito.version>
         <jamm.version>0.3.0</jamm.version>
-        <metrics.version>3.2.6</metrics.version>
+        <metrics.version>4.1.5</metrics.version>
```

Comment:
As described here: https://docs.datastax.com/en/developer/java-driver/3.5/manual/metrics/#metrics-4-compatibility
We need to add `metrics-jmx` also in out classpath. 
We are getting `NoClassDefFoundError: com/codahale/metrics/JmxReporter` here: https://travis-ci.org/github/JanusGraph/janusgraph/jobs/708630586
I think we can add `4.1.5` version of the required lib. I.e.:
```
<dependency>
    <groupId>io.dropwizard.metrics</groupId>
    <artifactId>metrics-jmx</artifactId>
    <version>${metrics.version}</version>
</dependency>
```

Also, as metrics update indirectly reflects cql, I would use `[full build]` in the comment to be sure that those tests don't fail.

Reply:
done let's hope.

Reply:
Also, I guess, we need to use `withoutJMXReporting()` when constructing CQL client right now as described in the upgrade guide. It says that if we want to use JmxReporter we need to use it in the following way:
```
Cluster cluster = Cluster.builder()
        .withoutJMXReporting()
        .build();
JmxReporter reporter =
    JmxReporter.forRegistry(cluster.getMetrics().getRegistry())
        .inDomain(cluster.getClusterName() + "-metrics")
        .build();
reporter.start();
```
I guess, if user sets a configuration property `metrics.jmx.enabled=true` we need to enable JmxReporter, otherwise we don't need to do it.
My assumption is that we need always use `builder.withoutJMXReporting()` when contrasting CQL `Cluster`. After CQL cluster is build, I think we could use some logic to start JmxReporter. I am not sure here and didn't test it, but I think something like the following code could be used:
```
private void initializeJMXReporterIfEnabled(final Configuration configuration) throws PermanentBackendException {
    if(configuration.get(METRICS_JMX_ENABLED)){
        if(!configuration.has(METRICS_JMX_DOMAIN)){
            throw new PermanentBackendException("Error initialising JMX reporter. JMX reporter domain isn't specified.");
        }
        JmxReporter.Builder jmxBuilder = JmxReporter.forRegistry(cluster.getMetrics().getRegistry())
            .inDomain(configuration.get(METRICS_JMX_DOMAIN));
        if(configuration.has(METRICS_JMX_AGENTID)){
            ArrayList<MBeanServer> mBeanServers = MBeanServerFactory.findMBeanServer(configuration.get(METRICS_JMX_AGENTID));
            if(mBeanServers.isEmpty()){
                throw new PermanentBackendException("Didn't find mBeanServer with id "+configuration.get(METRICS_JMX_AGENTID));
            } else if(mBeanServers.size()>1){
                throw new PermanentBackendException("Found multiple mBeanServers with id "
                    +configuration.get(METRICS_JMX_AGENTID)+" but had to find only a single mBeanServer. "+
                    "The next mBeanServers are found: "
                    +mBeanServers.stream().map(Object::toString).collect(Collectors.joining(", ")));
            }
            jmxBuilder.registerWith(mBeanServers.get(0));
        }
        JmxReporter reporter = jmxBuilder.build();
        reporter.start();
    }
}
```
That said, I didn't test if the above code works and didn't work with JmxReporter, so can't be sure on this.

Reply:
I think what be better to deactivate it and in your pr we use the newer client. This would leader that your PR would be a blocker for 0.6 which is okay for me.

Reply:
After a little bit of research, I think the above code can be simplified because we are already registering JMXReporter if it is enabled.
1) While building a CQL Cluster we need to add:
```
builder.withoutJMXReporting();
```
2) After the cluster is build we need to register CQL cluster metrics with our JMXReporter:
```
MetricManager.INSTANCE.getRegistry().registerAll(cluster.getMetrics().getRegistry());
```

I guess that's it but again, I didn't test it.

Reply:
@farodin91 I'm OK with introducing a blocker but I would suggest to try the next approach before introducing a blocker: https://github.com/JanusGraph/janusgraph/pull/2170#discussion_r455692818

If that works, we will be able to merge this PR without introducing a blocker

Reply:
I added some lines to cql driver.

Reply:
Will wait for TravisCI to complete. If all tests pass, I will test this PR internally also.

Reply:
You can't disable in the hadoop section easily jmx

Reply:
We could move org.janusgraph.hadoop for cql into its own project and try shading.

Reply:
I found a possible solution.

Reply:
You are genius. I even didn't think to include `JmxReporter` into the classpath. Tested locally `CQLScanJobIT` - all 3 tests passed.